File: Backend/IPC2Proyecto2/backendRevistas/src/main/java/Modelo/Revista.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Modelo;


/**
 *
 * @author ordson
 */
public class Revista {
    private int codigo;
    private String nombre;
    private String descripcion;
    private String fecha;
    private int autor;
    private Double precio;

    public Revista(int codigo, String nombre, String descripcion, String fecha, int autor, Double precio) {
        this.codigo = codigo;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.fecha = fecha;
        this.autor = autor;
        this.precio = precio;
    }

    public Revista(String nombre, String descripcion, String fecha, int autor, Double precio) {
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.fecha = fecha;
        this.autor = autor;
        this.precio = precio;
    }

    public Revista() {
    }

    public int getCodigo() {
        return codigo;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public String getFecha() {
        return fecha;
    }

    public void setFecha(String fecha) {
        this.fecha = fecha;
    }

    public int getAutor() {
        return autor;
    }

    public void setAutor(int autor) {
        this.autor = autor;
    }

    public Double getPrecio() {
        return precio;
    }

    public void setPrecio(Double precio) {
        this.precio = precio;
    }

    @Override
    public String toString() {
        return "Revista{" + "codigo=" + codigo + ", nombre=" + nombre + ", descripcion=" + descripcion + ", fecha=" + fecha + ", autor=" + autor + ", precio=" + precio + '}';
    }
    
    
}

File: Frontend/IPC2Proyecto2/src/app/objetos/Revista.ts
export class Revista{
    _codigo!:number;
    _nombre:String;
    _fecha:String;
    _precio:number;
    _descripcion:String;
    _autor:String;

    constructor(nombre: String, precio:number, fecha:String, descripcion:String, autor:String){
    this._nombre=nombre;
    this._precio= precio;
    this._fecha = fecha;
    this._descripcion = descripcion;
    this._autor = autor;
    }

    set nombre(nombre:String){
        this.nombre = nombre;
    }

    get nombre(){
        return this.nombre;
    }

    set precio(precio:number){
        this._precio = precio;
    }

    get precio(){
        return this._precio;
    }
    set codigo(codigo:number){
        this._codigo = codigo;
    }
    get fecha(){
        return this._fecha;
    }
    set fecha(fecha:String){
        this.fecha = fecha;
    }

    get descripcion(){
        return this._descripcion;
    }
    set descripcion(descripcion:String){
        this._descripcion = descripcion;
    }
    get autor(){
        return this._autor;
    }
    set autor(autor:String){
        this._autor = autor;
    }
    get codigo(){
        return this._codigo;
    }
}